app/components/qr/image-preview.tsx

```diff
@@ -0,0 +1,152 @@
+import {
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+  forwardRef,
+} from "react";
+
+export interface ImagePreviewProps {
+  qr: string;
+  logo: string;
+  size: "cable" | "small" | "medium" | "large";
+}
+
+export interface ImagePreviewState {
+  fontSize: number;
+  canvasSize: number;
+  logoSize: number;
+  qrImg: HTMLImageElement | null;
+  logoImg: HTMLImageElement | null;
+}
+
+export interface ImagePreviewRef {
+  exportToPNG(): string;
+}
+
+// NOTE: Do not trim the space at the end.
+// It used for spacing between the logo and the text.
+const PROPERTY_OF = "Property of  ";
+const FONT_SIZE_MAP = {
+  cable: 0,
+  small: 12,
+  medium: 15,
+  large: 18,
+};
+const LOGO_SIZE_MAP = {
+  cable: 0,
+  small: 24,
+  medium: 28,
+  large: 32,
+};
+
+const loadImage = (src: string): Promise<HTMLImageElement | null> =>
+  new Promise((resolve, reject) => {
+    if (src) {
+      const img = new Image();
+      img.src = src;
+      img.onload = () => {
+        resolve(img);
+      };
+      img.onerror = reject;
+    } else {
+      resolve(null);
+    }
+  });
+
+export const ImagePreview = forwardRef<ImagePreviewRef, ImagePreviewProps>(
+  function ImagePreview({ qr, size, logo }: ImagePreviewProps, ref) {
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+    const [state, setState] = useState({
+      fontSize: 15,
+      canvasSize: 190,
+      qrImg: null,
+      logoImg: null,
+    } as ImagePreviewState);
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        exportToPNG() {
+          const canvas = canvasRef.current;
+          return canvas?.toDataURL("image/png") ?? "";
+        },
+      }),
+      [canvasRef]
+    );
+
+    useEffect(() => {
+      const loadImages = Promise.all([qr, logo].map(loadImage));
+      loadImages.then(([qrImg, logoImg]) => {
+        if (!qrImg) return;
+
+        size === "cable" || !logo
+          ? setState({
+              qrImg,
+              logoImg,
+              fontSize: 0,
+              logoSize: 0,
+              canvasSize: qrImg.naturalHeight,
+            })
+          : setState({
+              qrImg,
+              logoImg,
+              fontSize: FONT_SIZE_MAP[size],
```

Comment:
I am checking this on desktop and it looks very big font, compared to the rest. Also I dont think it should be bold. Do you have a link to the Figma file? I can't find it so I can't send it but there you should be able to see the correct font-sizes. @jurrejansen can you help.

Reply:
I don't have the Figma link. But it should be fine with the new sizes I defined.
See samples below.


![arch-small-shelf-qr-code-cll5e7zqb0002d2fm3xt0qw6r](https://github.com/Shelf-nu/shelf.nu/assets/64963758/0f44e9a9-9ea8-459f-bdf1-46231e84f5b5)

![arch-medium-shelf-qr-code-cll5e7zqb0002d2fm3xt0qw6r](https://github.com/Shelf-nu/shelf.nu/assets/64963758/8d55f0b9-43c6-4e44-ba4a-4be48e5f1e2b)

![arch-large-shelf-qr-code-cll5e7zqb0002d2fm3xt0qw6r](https://github.com/Shelf-nu/shelf.nu/assets/64963758/1a337996-2103-4957-81c2-d0f422a6001b)

Reply:
Thanks @tobihans . We will make sure to include a figma link in the future.